corehq/apps/users/models_sql.py

```diff
@@ -179,6 +184,41 @@ class RoleAssignableBy(models.Model):
     )
 
 
+class UserUpdateMeta(jsonobject.JsonObject):
+    updated_via = jsonobject.StringProperty()
+
+
+class UpdateDetails(jsonobject.JsonObject):
+    meta = jsonobject.ObjectProperty(UserUpdateMeta)
+    changes = jsonobject.DictProperty()
+
+
+class HQLogEntry(models.Model):
+    """
+    HQ Adaptation of Django's LogEntry model
+    """
+    CREATE = 1
+    UPDATE = 2
+    DELETE = 3
+
+    ACTION_FLAG_CHOICES = (
+        (CREATE, _('Create')),
+        (UPDATE, _('Update')),
+        (DELETE, _('Delete')),
+    )
+    domain = models.CharField(max_length=255, db_index=True)
+    object_type = models.CharField(max_length=255, db_index=True, choices=(
+        ('CommCareUser', 'CommCareUser'),
+        ('WebUser', 'WebUser'),
+    ))
+    object_id = models.CharField(max_length=128, db_index=True)
+    by_user_id = models.CharField(max_length=128, db_index=True)
+    details = JSONField(default=dict)  # UpdateDetails
+    message = models.TextField(_('change message'), blank=True)
```

Comment:
why are these verbose names necessary? Do they appear anywhere on the UI? They also don't seem to offer much more than the field name other than potential translation.

Reply:
I am still fine tuning this. First step was to make the current implementation work with the new model, which i am still working on and hence the "message" field. I am not sure if it will be retained eventually or not, once details is populated with user fields and meta information is moved to it as well, this should not be needed.
For now, just picked the definition of "message" from django's LogEntry.

Reply:
verbose names removed.